canvas_modules/common-canvas/src/common-properties/components/virtualized-grid/virtualized-grid.jsx

```diff
@@ -0,0 +1,476 @@
+		const translatedHeaderCheckboxLabel = props.intl.formatMessage(
+			{ id: "virtualizedTable.header.checkbox.label", defaultMessage: defaultMessages["virtualizedTable.header.checkbox.label"] },
+			{ header_checkbox_label: headerCheckboxLabel }
+		);
+		const checkbox = props.selectable && props.rowSelection !== ROW_SELECTION.SINGLE
+			? (<th className="properties-vt-column properties-vt-header-checkbox">
+				<Checkbox
+					id={`properties-vt-hd-cb-${uuid}-${props.scrollKey}`}
+					onChange={setAllRowsSelected}
+					checked={props.checkedAll}
+					labelText={translatedHeaderCheckboxLabel}
+					hideLabel
+					readOnly={props.readOnly}
+				/>
+			</th>)
+			: "";
+		return checkbox;
+	};
+
+	const renderDataRowCheckbox = (rowIndex, rowData) => {
+		if (props.summaryTable) {
+			return <div className="properties-vt-row-checkbox" />;
+		}
+
+		let selectOption = "";
+		if (props.selectable && props.rowSelection !== ROW_SELECTION.SINGLE) {
+			const rowSelected = isRowSelected(rowData.originalRowIndex); // use current row index when Sorted
+			const rowDisabled = typeof rowData.disabled === "boolean" ? rowData.disabled : false;
+			const translatedRowCheckboxLabel = props.intl.formatMessage(
+				{ id: "virtualizedTable.row.checkbox.label", defaultMessage: defaultMessages["virtualizedTable.row.checkbox.label"] },
+				{ row_index: rowIndex + 1, table_label: (props.tableLabel ? props.tableLabel : "") }
+			);
+
+			selectOption = (<td className="properties-vt-row-checkbox"
+				onMouseEnter={(evt) => overSelectOption(evt)}
+				onMouseLeave={(evt) => overSelectOption(evt)}
+				onFocus={(evt) => overSelectOption(evt)}
+				onBlur={(evt) => overSelectOption(evt)}
+				onKeyDown={(evt) => {
+					if (evt.code === "Space" || evt.code === "Enter") {
+						onRowClick(evt, rowIndex, rowData);
+					}
+				}}
+			>
+				<Checkbox
+					id={`properties-vt-row-cb-${uuid}-${props.scrollKey}-${rowIndex}`}
+					key={`properties-vt-row-cb-${props.scrollKey}-${rowIndex}`}
+					labelText={translatedRowCheckboxLabel}
+					hideLabel
+					checked={rowSelected}
+					disabled={rowDisabled}
+					readOnly={props.readOnly}
+				/>
+			</td>);
+		}
+		return selectOption;
+	};
+
+	const overSelectOption = (evt) => {
+		// Incase of readonly table disable all events
+		if (!props.readOnly) {
+			// Differentiate between mouse and keyboard event
+			if (evt.type === "mouseenter" && !keyBoardEventCalled) {
+				setMouseEventCalledState(true);
+				setIsOverSelectOptionState(true);
+			} else if (evt.type === "mouseleave" && mouseEventCalled) {
+				setMouseEventCalledState(false);
+				setIsOverSelectOptionState(false);
+			} else if (evt.type === "focus" && !mouseEventCalled) {
+				setKeyBoardEventCalledState(true);
+				setIsOverSelectOptionState(!isOverSelectOption);
+			} else if (evt.type === "blur" && keyBoardEventCalled) {
```

Comment:
@nmgokhale Do you remember what the evt type `focus` and `blur` does?
I dont see why `isOverSelectOption` should be toggled when testing in the old table
![overSelectOption](https://github.com/user-attachments/assets/7ed842a0-8866-482f-8d9c-79efadf7ef9f)

Reply:
`Focus` and `blur` events are called when using `tab/shift+tab` keys from keyboard.
It was added in this PR - https://github.com/elyra-ai/canvas/pull/438

I think it was to select checkbox using keyboard - https://github.com/elyra-ai/canvas/blob/d5bada0fc486dcbcc7191318fa45bcab6db8f766/canvas_modules/common-canvas/__tests__/_utils_/table-utils.js#L153